manager/src/main/java/org/openremote/manager/mqtt/ActiveMQORSecurityManager.java

```diff
@@ -98,13 +112,84 @@ protected Subject getAuthenticatedSubject(String user,
+            List<X509Certificate> clientAuthCerts = Arrays.stream(certs).filter(e -> {
+                try {
+                    return e.getExtendedKeyUsage().contains(CLIENT_AUTH_EKU_OID);
+                } catch (CertificateParsingException ex) {
+                    throw new RuntimeException(ex);
+                } catch (NullPointerException ignored){return false;}
+            }).toList();
+
+            if (clientAuthCerts.size() != 1) {
+                String errMsg = "Presented certificate chain contains " + clientAuthCerts.size() +
+                        " certificates with Client Authentication Extended Key Usage. " +
+                        "Expected exactly 1 certificate. Please provide a valid certificate chain. "+ connectionToString(remotingConnection);
+                LOG.log(Level.SEVERE, errMsg);
+                throw new LoginException(errMsg);
+            }
+
+            X509Certificate leaf = clientAuthCerts.getFirst();
+
+            String dn = leaf.getSubjectX500Principal().getName();
+            try {
+                LdapName ldapName = new LdapName(dn);
+                for (Rdn rdn : ldapName.getRdns()) {
+                    String type = rdn.getType();
+                    String value = rdn.getValue().toString();
+                    if (realm == null && "OU".equalsIgnoreCase(type)) {
+                        realm = value;
+                    } else if (TextUtil.isNullOrEmpty(user) && "CN".equalsIgnoreCase(type)) {
+                        user = value;
+                    }
+                }
+            } catch (InvalidNameException e) {
+                LOG.log(Level.FINE, "Failed to parse subject DN from client certificate: " + dn, e);
+            } catch (Exception e) {
+                LOG.log(Level.SEVERE, "Failed to process given client certificate");
+            }
+
+            if(realm == null){
+                LOG.log(Level.INFO, "Client certificate was provided, but no realm found in certificate subject. " +
+                        "Falling back to the username as a realm name. Subject DN=" + dn);
+
+                if (identityProvider.getRealm(user) != null) {
+                    realm = user;
+                }else{
+                    LOG.log(Level.WARNING, "No realm found matching the username. " +
+                            "Failing login attempt. Subject DN=" + dn);
+                    throw new LoginException("No realm found matching the username from client certificate.");
+                }
+            }
+
+            if(!TextUtil.isNullOrEmpty(user)) {
+                User dbUser = null;
+                User[] users = identityProvider.queryUsers(new UserQuery().usernames(new StringPredicate(SERVICE_ACCOUNT_PREFIX + user).match(AssetQuery.Match.EXACT)));
+                if (users != null && users.length == 1) {
+                    dbUser = users[0];
+                }else if(users != null && users.length > 1) {
```

Comment:
Inconsistent code style, please use spaces after `if/ese` statement and after curly braces